SNEWS_PT/snews_pub.py

```diff
@@ -76,91 +77,42 @@ def display_message(self, message):
-            messages.append(sig_message)
-
-        # TimingTier if timing_series exists (@Seb why do we need p_value to be float?)
-        if type(self.timing_series) == list and type(self.p_value) == float:
-            data = snews_pt_utils.time_tier_data(machine_time=self.machine_time,
-                                                 nu_time=self.nu_time,
-                                                 timing_series=self.timing_series,
-                                                 p_val=self.p_value,
-                                                 meta=meta
-                                                 )
-
-            time_message = self.schema.get_schema(tier='TimeTier', data=data, )
-            messages.append(time_message)
-
-        # Retraction Command if retraction field is passed
-        if self.n_retract_latest != 0 and type(self.which_tier) == str:
-            data = snews_pt_utils.retraction_data(machine_time=self.machine_time,
-                                                  which_tier=self.which_tier, n_retract_latest=self.n_retract_latest,
-                                                  retraction_reason=self.retraction_reason, meta=meta)
-            retraction_message = self.schema.get_schema(tier='Retraction', data=data, )
-            messages.append(retraction_message)
-
-        # Heartbeat if detector status passed (and maybe other==None ?)
-        # they can also set status=ON when they submit coincidence, do we want to publish also a HB at the same time?
-        if type(self.detector_status) == str and type(self.machine_time) == str:
-            data = snews_pt_utils.heartbeat_data(detector_status=self.detector_status, machine_time=self.machine_time)
-            heartbeat_message = self.schema.get_schema(tier='Heartbeat', data=data, )
-            messages.append(heartbeat_message)
-
-        return messages
+    """
+    To use a json file call SNEWSTiersPublisher().from_json(<filename>)
+    Else, the following keys and more kwargs can be passed
+      `detector_name`
+      `machine_time`
+      `nu_time`
+      `p_val`
+      `p_values`
+      `timing_series`
+      `which_tier`
+      `n_retract_latest`
+      `retraction_reason`
+      `detector_status`
+      `is_pre_sn`
+    """
+    def __init__(self, env_file=None, **kwargs):
+        snews_pt_utils.set_env(env_file)
+        self.args_dict = dict(**kwargs)
+        # if is_pre_sn not given, it is False. If name not given fetch from env
+        self.args_dict["is_pre_sn"] = kwargs.get("is_pre_sn", False)
+        self.args_dict['detector_name'] = kwargs.get("detector_name", os.getenv('DETECTOR_NAME'))
+        self.schema = Message_Schema(detector_key=self.args_dict['detector_name'], is_pre_sn=kwargs.get("is_pre_sn", False))
+        self.messages, self.tiernames = snews_pt_utils._tier_decider(self.args_dict)
+
+        # if we want to have all of them as attributes,
+        # for key, value in kwargs.items():
```

Comment:
here we can actually add everything as individual attributes this is equal to e.g.
`self.p_values = args['p_values']`
but we already have the `self.args_dict`, and also individual formatted messages (and what tier they correspond to)